pep/unmarshal.go

```diff
@@ -37,29 +37,15 @@ var (
 	}
 )
 
-func fillResponse(res pb.Msg, v interface{}) error {
-	switch v := v.(type) {
-	case *pb.Msg:
-		*v = res
-		return nil
-
-	case *pdp.Response:
-		if v.Obligations == nil {
-			effect, a, err := pdp.UnmarshalResponseAssignments(res.Body)
-			if err != nil {
-				if _, ok := err.(*pdp.ResponseServerError); !ok {
-					return err
-				}
-			}
+type CachedResponse struct {
```

Comment:
The name `CachedResponse` tells that response goes from cache. I'd name it `ResponseEx` (as MS Windows usually does for structures with extra flags).

Reply:
Seem to recall seeing those when I last work with the WIN32 API - some 20 years ago...

From your earlier review comment though, we don't need to introduce a new struct anyway.

Reply:
😀